### DISCUSSION TITLE
Ukrainian numbers (+38) are not in the list of two-factor authentication

### DISCUSSION THREAD

**Discussion by Glebuar** [community/community#14416] [Other Feature Feedback, Questions, & Ideas]
How to set up two-factor authentication using SMS for Ukraine?

**Comment by queenofcorgis** [comment]
Hi there - Ukraine is currently not on our list of supported countries [per this list](https://docs.github.com/en/enterprise-cloud@latest/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported).

**Reply by ichepurnoy** [reply]
Ukraine is currently not on this list, as per that link, because of 'SMS Delivery success rates'. But every other service including Google has been delivering their SMS without any problem, for more than a decade.
Uganda though _is_ on the list 😀 
Github, please reconsider this (looking like politically-biased) decision.

**Reply by suprstarrd** [reply]
Russia is also not on the list. Doesn't seem politically biased to me

**Reply by flysubmarina** [reply]
> Russia is also not on the list. Doesn't seem politically biased to me

Anyway, we need it, its not political topic i think, but it will be useful for our developers

**Reply by YuriiYurchyk** [reply]
> Ukraine is currently not on this list, as per that link, because of 'SMS Delivery success rates'. But every other service including Google has been delivering their SMS without any problem, for more than a decade. Uganda though _is_ on the list 😀 Github, please reconsider this (looking like politically-biased) decision.

If suddenly the problem is in the price of delivery or its quality, then GitHub can offer to make two-factor authentication for Ukraine paid and for users to pay for SMS, I don’t know who exactly to address this message to, I’ll just leave it here

**Comment by rungrotArunon** [comment]
Thanks you

**Comment by flysubmarina** [comment]
Please, fix it. Ukrainian developers need it. For me, it will be helpful if i can set up a fallback SMS number or use TFA via SMS

**Comment by yyaremenko** [comment]


**Reply by suprstarrd** [reply]
...0, because Russia also doesn't have SMS 2FA

**Comment by yakhoruzhenko** [comment]
Discriminating Ukrainian developers who are trying to work despite the war and the absence of electricity. Nice move buddies.

**Comment by BrettskiPy** [comment]
As an American company, this is your time to step up and do the right thing. Our engineer friends in Ukraine are asking for your support. Please prioritize this and ensure equal access. Ukraine is fighting for their freedom and to keep themselves warm... They don't need more problems.

**Reply by yakhoruzhenko** [reply]
Thank you for the support! Personal and and of course on the state level.

**Reply by poratuk** [reply]
Its wery warm from your side. Thanks @BrettskiPy

**Comment by vladyslav2** [comment]
> SMS Delivery success rates

Is it possible to get more details here?
mobile network working very well in Ukraine, despite electricity and heat challenges

**Comment by queenofcorgis** [comment]
GitHub is aware of concerns expressed by community members regarding SMS country availability and is actively reviewing this. We do not have additional updates available at this time.

If updates are made they will be available here: [Countries where SMS authentication is supported](https://docs.github.com/en/enterprise-cloud@latest/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported).  Thanks for sharing your comments and feedback

**Reply by BrettskiPy** [reply]
Thank you for the follow up and I appreciate that GitHub is actively reviewing this.

**Comment by RONIE70** [comment]
> How to set up two-factor authentication using SMS for Ukraine?

Generar un código a través de una aplicación TOTP
Si decides configurar una autenticación de dos factores mediante una aplicación TOTP en tu smartphone, puedes generar un código de autenticación para GitHub en cualquier momento. En la mayoría de los casos, el lanzamiento de la aplicación generará un código nuevo. Deberías consultar la documentación de la aplicación para conocer las instrucciones específicas.

Si eliminas las aplicaciones móviles después de configurar la autenticación de dos factores, deberás proporcionar tu código de recuperación para obtener acceso a tu cuenta. Para más información, vea "[Recuperación de la cuenta en caso de perder las credenciales de autenticación en dos fases](https://docs.github.com/es/articles/recovering-your-account-if-you-lose-your-2fa-credentials)"

Recibir un mensaje de texto
Si configuras una autenticación de dos factores mediante mensajes de texto, GitHub te enviará un mensaje de texto con tu código de autenticación.

Verificar con GitHub Mobile
Si instalaste e iniciaste sesión en GitHub Mobile, podrías elegir autenticarte con GitHub Mobile para la autenticación bifactorial.

Inicia sesión en GitHub con tu buscador, utilizando tu usuario y contraseña.

Si agregaste una llave de seguridad a tu cuenta, primero se te pedirá insertar y utilizar una llave de seguridad. Para omitir el uso de una clave de seguridad, haga clic en Autenticar con GitHub Mobile .
Desafío de la autenticación en dos fases en GitHub con "Autenticación con GitHub Mobile" resaltado

GitHub te enviará una notificación push para verificar tu intento de inicio de sesión. Abrir la notificación push o abrir la app de GitHub Mobile mostrará un mensaje que te pide aprobar o rechazar este intento de inicio de sesión.

Nota: Este mensaje podría indicarle que escriba un número de dos dígitos que se muestra dentro del explorador en el que ha iniciado sesión.

El reto de autenticación bifactorial con GitHub Mobile que requiere una entrada de dos dígitos

Cuando apruebes el intento de inicio de sesión utilizando GitHub Mobile, tu buscador completará el inicio de sesión automáticamente.
Si rechazas el intento de inicio de sesión, se prevendrá la finalización de la autenticación. Para más información, vea "[Protección de la cuenta y los datos](https://docs.github.com/es/authentication/keeping-your-account-and-data-secure)".
Usar autenticación de dos factores con la línea de comando
Después de habilitar 2FA, ya no usarás la contraseña para acceder a GitHub en la línea de comandos. En vez de ello tendrás que usar el Administrador de credenciales de Git, un token de acceso personal o una clave SSH.

Autenticación en la línea de comandos mediante el Administrador de credenciales de Git
El [Administrador de credenciales de Git](https://github.com/GitCredentialManager/git-credential-manager/blob/main/README.md) es un asistente de credenciales de Git seguro que se ejecuta en Windows, macOS y Linux. Para obtener más información sobre los asistentes de credenciales de Git, consulta [Evitar repeticiones](https://git-scm.com/docs/gitcredentials#_avoiding_repetition) en el libro de Git de Pro.

Las instrucciones de configuración varían en función del sistema operativo del equipo. Para obtener más información, consulta [Descargar e instalar](https://github.com/GitCredentialManager/git-credential-manager/blob/main/README.md#download-and-install) en el repositorio GitCredentialManager/git-credential-manager.

Autenticar en la línea de comando mediante HTTPS
Después de haber habilitado 2FA, debes crear un token de acceso personal para usar una contraseña al autenticar a GitHub en la línea de comando mediante las URL HTTPS.

Cuando se te solicite el nombre de usuario y la contraseña en la línea de comando, usa tu nombre de usuario GitHub y el token de acceso personal. La indicación de la línea de comando no especificará que debes ingresar tu token de acceso personal cuando se te solicite la contraseña.

Para más información, vea "[Creación de un token de acceso personal](https://docs.github.com/es/github/authenticating-to-github/creating-a-personal-access-token)".

Autenticar en la línea de comandos mediante SSH
La habilitación de 2FA no cambia el modo de autenticar a GitHub en la línea de comando mediante las URL SSH. Para más información sobre cómo configurar y usar una clave SSH, vea "[Conexión a GitHub con SSH](https://docs.github.com/es/articles/connecting-to-github-with-ssh)".

Usar autenticación de dos factores para acceder a un repositorio mediante Subversion
Cuando accedas a un repositorio mediante Subversion, debes proporcionar un token de acceso personal en lugar de ingresar tu contraseña. Para más información, vea "[Creación de un token de acceso personal](https://docs.github.com/es/github/authenticating-to-github/creating-a-personal-access-token)".

Solución de problemas
Si pierdes el acceso a tus credenciales de autenticación de dos factores, puedes usar tus códigos de recuperación u otro método de recuperación (si has configurado uno) para recuperar el acceso a tu cuenta. Para más información, vea "[Recuperación de la cuenta si se pierden las credenciales de 2FA](https://docs.github.com/es/articles/recovering-your-account-if-you-lose-your-2fa-credentials)".

Si tu autenticación falla varias veces, es posible que desees sincronizar el reloj de tu teléfono con tu proveedor móvil. Frecuentemente, esto involucra la verificación de la opción "Establecer automáticamente" en el reloj de tu teléfono, en lugar de brindar tu propia zona horaria.

**Comment by RONIE70** [comment]
> How to set up two-factor authentication using SMS for Ukraine?

Generar un código a través de una aplicación TOTP
Si decides configurar una autenticación de dos factores mediante una aplicación TOTP en tu smartphone, puedes generar un código de autenticación para GitHub en cualquier momento. En la mayoría de los casos, el lanzamiento de la aplicación generará un código nuevo. Deberías consultar la documentación de la aplicación para conocer las instrucciones específicas.

Si eliminas las aplicaciones móviles después de configurar la autenticación de dos factores, deberás proporcionar tu código de recuperación para obtener acceso a tu cuenta. Para más información, vea "[Recuperación de la cuenta en caso de perder las credenciales de autenticación en dos fases](https://docs.github.com/es/articles/recovering-your-account-if-you-lose-your-2fa-credentials)"

Recibir un mensaje de texto
Si configuras una autenticación de dos factores mediante mensajes de texto, GitHub te enviará un mensaje de texto con tu código de autenticación.

Verificar con GitHub Mobile
Si instalaste e iniciaste sesión en GitHub Mobile, podrías elegir autenticarte con GitHub Mobile para la autenticación bifactorial.

Inicia sesión en GitHub con tu buscador, utilizando tu usuario y contraseña.

Si agregaste una llave de seguridad a tu cuenta, primero se te pedirá insertar y utilizar una llave de seguridad. Para omitir el uso de una clave de seguridad, haga clic en Autenticar con GitHub Mobile .
Desafío de la autenticación en dos fases en GitHub con "Autenticación con GitHub Mobile" resaltado

GitHub te enviará una notificación push para verificar tu intento de inicio de sesión. Abrir la notificación push o abrir la app de GitHub Mobile mostrará un mensaje que te pide aprobar o rechazar este intento de inicio de sesión.

Nota: Este mensaje podría indicarle que escriba un número de dos dígitos que se muestra dentro del explorador en el que ha iniciado sesión.

El reto de autenticación bifactorial con GitHub Mobile que requiere una entrada de dos dígitos

Cuando apruebes el intento de inicio de sesión utilizando GitHub Mobile, tu buscador completará el inicio de sesión automáticamente.
Si rechazas el intento de inicio de sesión, se prevendrá la finalización de la autenticación. Para más información, vea "[Protección de la cuenta y los datos](https://docs.github.com/es/authentication/keeping-your-account-and-data-secure)".
Usar autenticación de dos factores con la línea de comando
Después de habilitar 2FA, ya no usarás la contraseña para acceder a GitHub en la línea de comandos. En vez de ello tendrás que usar el Administrador de credenciales de Git, un token de acceso personal o una clave SSH.

Autenticación en la línea de comandos mediante el Administrador de credenciales de Git
El [Administrador de credenciales de Git](https://github.com/GitCredentialManager/git-credential-manager/blob/main/README.md) es un asistente de credenciales de Git seguro que se ejecuta en Windows, macOS y Linux. Para obtener más información sobre los asistentes de credenciales de Git, consulta [Evitar repeticiones](https://git-scm.com/docs/gitcredentials#_avoiding_repetition) en el libro de Git de Pro.

Las instrucciones de configuración varían en función del sistema operativo del equipo. Para obtener más información, consulta [Descargar e instalar](https://github.com/GitCredentialManager/git-credential-manager/blob/main/README.md#download-and-install) en el repositorio GitCredentialManager/git-credential-manager.

Autenticar en la línea de comando mediante HTTPS
Después de haber habilitado 2FA, debes crear un token de acceso personal para usar una contraseña al autenticar a GitHub en la línea de comando mediante las URL HTTPS.

Cuando se te solicite el nombre de usuario y la contraseña en la línea de comando, usa tu nombre de usuario GitHub y el token de acceso personal. La indicación de la línea de comando no especificará que debes ingresar tu token de acceso personal cuando se te solicite la contraseña.

Para más información, vea "[Creación de un token de acceso personal](https://docs.github.com/es/github/authenticating-to-github/creating-a-personal-access-token)".

Autenticar en la línea de comandos mediante SSH
La habilitación de 2FA no cambia el modo de autenticar a GitHub en la línea de comando mediante las URL SSH. Para más información sobre cómo configurar y usar una clave SSH, vea "[Conexión a GitHub con SSH](https://docs.github.com/es/articles/connecting-to-github-with-ssh)".

Usar autenticación de dos factores para acceder a un repositorio mediante Subversion
Cuando accedas a un repositorio mediante Subversion, debes proporcionar un token de acceso personal en lugar de ingresar tu contraseña. Para más información, vea "[Creación de un token de acceso personal](https://docs.github.com/es/github/authenticating-to-github/creating-a-personal-access-token)".

Solución de problemas
Si pierdes el acceso a tus credenciales de autenticación de dos factores, puedes usar tus códigos de recuperación u otro método de recuperación (si has configurado uno) para recuperar el acceso a tu cuenta. Para más información, vea "[Recuperación de la cuenta si se pierden las credenciales de 2FA](https://docs.github.com/es/articles/recovering-your-account-if-you-lose-your-2fa-credentials)".

Si tu autenticación falla varias veces, es posible que desees sincronizar el reloj de tu teléfono con tu proveedor móvil. Frecuentemente, esto involucra la verificación de la opción "Establecer automáticamente" en el reloj de tu teléfono, en lugar de brindar tu propia zona horaria.

**Comment by Patlatus** [comment]
well, how can you require to enable mandatory 2fa and not allow the option to setup sms 2fa for Ukrainian cell phone number?

I understand, I could potentially use some third-party mfa, but what if I lose my phone or access to that mfa app, would it mean that I would lose access to my github account completely? If I could attach to Ukrainian cell phone number and if I lost the sim card, I could still restore my phone number.

**Reply by queenofcorgis** [reply]
Hi there, SMS is just one of the options we have for 2FA: TOTP apps, security keys, and the GitHub Mobile app are all authentication methods we support.

**Comment by Iuriy-Budnikov** [comment]
That a shame for github!

**Comment by radetzkie** [comment]
Hi there.

So? Any updates on this ticket? 
Completely agree with this point: 

> I understand, I could potentially use some third-party mfa, but what if I lose my phone or access to that mfa app, would it mean that I would lose access to my github account completely? If I could attach to Ukrainian cell phone number and if I lost the sim card, I could still restore my phone number.

All services are sending SMS to UA without any issues (Google, Etsy, Apple, CloudAcademy, Microsoft, OpenAI - just to name a few I've used recently).

**Comment by ghost** [comment]
GitHub punishes Ukrainian people because they have bad neighbors. Bravo GitHub!
If SMS 2FA is unavailable to you just use another options. These options:

1. GitLab
2. Google Drive
3. OneDrive

etc.
Це зрада від GitHub!

**Comment by theanurin** [comment]
Funny....

Belarus (under sanctions and have bad neighbors) in the list of supported countries

@github - do you have any plan to support SMS auth in Ukraine?

**Comment by mr-addams** [comment]
it looks like " im pohuy" 
That a real shame for GitHub!

**Comment by amseledka** [comment]
no chance to user 2FA with Ukrainian phone number
 add Ukraine code(+380)!

**Comment by lepik1108** [comment]
Just add it yourself!

![Screenshot from 2024-10-07 06-29-41](https://github.com/user-attachments/assets/088c64e8-f6d7-4c03-be36-ea73dd07f189)

Tested. It works.
Applied some of my poor HTML editing skills to that SMS-verification form...

![photo_2024-10-07_05-34-13](https://github.com/user-attachments/assets/f40c80e9-a686-4b56-b2f4-3d194681405a)
![photo_2024-10-07_05-35-19](https://github.com/user-attachments/assets/15a66020-95f5-434e-bd47-aee08a855d53)

Just in case entered github bug-bounty program and found this:
https://bounty.github.com/ineligible.html#bypassing_country_restrictions_for_sms_twofactor_authentication
So I'm not violating github terms of service, not abusing a security vulnerability.

P.S.:
I've been working in IT for a while, as a Tech Support, QA(manual/automation/lead), Dev, DevOps and still not consider myself a Senior level Engineer sometimes. Don't care about that too much, just satisfied I can solo handle entire development process from scratch pretty effective.
Kinda sad, that majority of people, especially in big companies are as lazy and spoiled as amount of inappropriate posts above...
So, сolleagues, please stop complain, just do your job well, learn new stuff everyday and share your knowledge with others every time you can...

**Reply by GofSoftware** [reply]
Doesn't work for me, probably they already fixed this bug.

**Comment by PeacefulPieDev** [comment]
Still no updates. Very decision of Github.
